taca/analysis/analysis_nanopore.py

```diff
@@ -416,6 +367,159 @@ def ont_updatedb(ont_run):
+    df = pd.read_csv(glob_pore_activity[0])
+    df.sort_values(by="Experiment Time (minutes)", inplace=True)
+    df = df.pivot_table(
+        "State Time (samples)", "Experiment Time (minutes)", "Channel State"
+    )
+
+    # Use pore counts to calculate new metrics
+    df["all"] = df.sum(axis=1)
+    df["healthy"] = df.strand + df.adapter + df.pore
+    df["productive"] = df.strand + df.adapter
+
+    df["health"] = df["healthy"] / df["all"]
+    df["efficacy"] = df["productive"] / df["healthy"]
+
+    # Look at peaks within 1st hour of the run and define some metrics
+    df_h1 = df[0:60]
+    pore_activity["peak_pore_health_pc"] = round(
+        100 * float(df_h1.loc[df_h1.health == df_h1.health.max(), "health"]), 2
+    )
+    pore_activity["peak_pore_efficacy_pc"] = round(
+        100 * float(df_h1.loc[df_h1.efficacy == df_h1.efficacy.max(), "efficacy"]), 2
+    )
+
+    # Calculate the T90
+    # -- Get the cumulative sum of all productive pores
+    df["cum_productive"] = df["productive"].cumsum()
+    # -- Find the timepoint (h) at which the cumulative sum >= 90% of the absolute sum
+    t90_min = df[df["cum_productive"] >= 0.9 * df["productive"].sum()].index[0]
+    pore_activity["t90_h"] = round(t90_min / 60, 1)
+
+    # Add to the db update
+    db_update["pore_activity"] = pore_activity
+
+
+def parse_minknow_json(ont_run, db_update):
+    """Parse useful stuff from the MinKNOW .json report to add to CouchDB"""
+
+    logger.info(f"Parsing report JSON of run {ont_run.run_id}")
+
+    glob_report_json = glob.glob(ont_run.run_dir + "/report*.json")
+
+    if len(glob_report_json) == 0:
+        error_message = f"Run {ont_run.run_id} is marked as finished, but missing .json report file."
+        logger.error(error_message)
+        raise AssertionError(error_message)
+    elif len(glob_report_json) > 1:
+        error_message = f"Run {ont_run.run_id} is marked as finished, but contains conflicting .json report files."
+        logger.error(error_message)
+        raise AssertionError(error_message)
+
+    dict_json_report = json.load(open(glob_report_json[0], "r"))
+
+    # Initialize return dict
+    parsed_data = {}
+
+    # These sections of the .json can be added as they are
+    for section in [
+        "software_versions",
+        "host",
+        "protocol_run_info",
+        "user_messages",
+    ]:
+        parsed_data[section] = dict_json_report[section]
+
+    # Only parse the last acquisition section, which contains the actual sequencing data
+    seq_metadata = dict_json_report["acquisitions"][-1]
+    seq_metadata_trimmed = {}
+
+    # -- Run info subsection
+    seq_metadata_trimmed["acquisition_run_info"] = {}
+
+    seq_metadata_trimmed["acquisition_run_info"]["yield_summary"] = seq_metadata[
+        "acquisition_run_info"
+    ]["yield_summary"]
+
+    seq_metadata_trimmed["acquisition_run_info"]["yield_summary"] = seq_metadata[
+        "acquisition_run_info"
+    ]["yield_summary"]
```

Comment:
This looks the same to me?

Reply:
ec1b44e793aad7685e5d9b92946a427950d1ff16